### DISCUSSION TITLE
poetry show outdated from source [help wanted]

### DISCUSSION THREAD

**Discussion by Rishabh04-02** [python-poetry/poetry#9404] [Q&A]
I've two packages:

1. package1
2. package2

I'm installing it from my source:
```
[[tool.poetry.source]]
name = "mysource"
url = "https://example.com/mysource/simple/"
priority = "explicit"
```
pyproject.toml
```
[tool.poetry.dependencies]
package1 = { version = "==2.0.2", source = "mysource" }
package2 = {version = "==0.2.0", source = "mysource"}

[[tool.poetry.source]]
name = "mysource"
url = "https://example.com/mysource/simple/"
priority = "explicit"
```
Both of these packages are also available in pypi.org

so when i run the following command to check for outdated packages I get incorrect output
```
$ poetry show --outdated
package1             2.0.2                     28.0.0 description of package
package2             0.2.0                     5.8.0  description of package
```
It compared the installed versions with the versions of pypi.org packages, which is incorrect.

I don't want to change the priority of my source to primary as that is taking too long to run.

How can I check for outdated packages which are installed from mysource?

Expectation is to match the version from mysource and not from pypi.org
```
$ poetry show --outdated
package1             2.0.2                     2.0.3 description of package
package2             0.2.0                     0.2.1  description of package
```

question source: https://stackoverflow.com/questions/78454417/poetry-show-outdated-from-source

**Comment by Secrus** [comment]
I think it's a bug in `show` command that it doesn't take source into consideration when `--outdated`.
